### PR TITLE
Add variable size binary array

### PR DESCRIPTION
**Pull request by Alex-PLACET**

### CODE REVIEW

File: include/sparrow_v01/layout/variable_size_binary_array.hpp [hunk 1]
```diff
@@ -0,0 +1,594 @@
...
+        // This is to avoid const char* from begin caught by the previous
+        // operator= overload. It would convert const char* to const char[N],
+        // including the null-terminating char.
+        template <class U = typename L::inner_value_type>
+            requires std::assignable_from<U&, const char*>
+        self_type& operator=(const char* rhs);
+
+        size_type size() const;
+
+        // iterator begin();
```
Comment (Alex-PLACET) [inline include/sparrow_v01/layout/variable_size_binary_array.hpp:148]: I commented the "mutable" parts. Tell me if you prefer that I just remove the code

File: include/sparrow/arrow_interface/arrow_array_schema_info_utils.hpp [hunk 2]
```diff
@@ -91,9 +94,46 @@ namespace sparrow
...
+        buffer_type::OFFSETS_64BIT,
+        buffer_type::SIZES_64BIT
+    };
+    constexpr std::array<buffer_type, 1> buffers_types_validity{buffer_type::VALIDITY};
+    constexpr std::array<buffer_type, 1> buffers_types_types_ids{buffer_type::TYPE_IDS};
+    constexpr std::array<buffer_type, 2> buffers_types_types_ids_offset32{
+        buffer_type::TYPE_IDS,
+        buffer_type::OFFSETS_32BIT
+    };
+    constexpr std::array<buffer_type, 0> buffers_types_empty{};
```
Comment (DerThorsten) [inline include/sparrow/arrow_interface/arrow_array_schema_info_utils.hpp:132]: I dont like this

**Review by DerThorsten — DISMISSED** [review]

File: include/sparrow/arrow_interface/arrow_array_schema_info_utils.hpp (hunk 2 above)
Comment (DerThorsten) [reply include/sparrow/arrow_interface/arrow_array_schema_info_utils.hpp:132]: having this exhaustive amount of variables with long unreadable names  is hard to read and a code smell IMO

File: include/sparrow/arrow_array_schema_proxy.hpp [hunk 3]
```diff
@@ -229,7 +235,9 @@ namespace sparrow
         std::vector<arrow_proxy> m_children;
         std::unique_ptr<arrow_proxy> m_dictionary;
 
-        struct impl_tag{};
+        struct impl_tag
+        {
+        };
```
[the remark this answers is not in the document]
Comment (Alex-PLACET) [reply include/sparrow/arrow_array_schema_proxy.hpp:240]: It's used in the constructors l.356 to l.367

File: include/sparrow_v01/layout/variable_size_binary_array.hpp (hunk 1 above)
Comment (Alex-PLACET) [reply include/sparrow_v01/layout/variable_size_binary_array.hpp:148]: Let's keep it if it don't bother you.

File: include/sparrow/arrow_interface/arrow_array_schema_info_utils.hpp (hunk 2 above)
Comment (Alex-PLACET) [reply include/sparrow/arrow_interface/arrow_array_schema_info_utils.hpp]: Done 👍

File: include/sparrow/arrow_array_schema_proxy.hpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (JohanMabille) [reply include/sparrow/arrow_array_schema_proxy.hpp:240]: Indeed, I missed it, sorry for that.

**Review by JohanMabille — APPROVED** [review]

**Pull request merged**